doc/userguide/devguide/codebase/installation-from-git.rst

```diff
@@ -0,0 +1,165 @@
+Installation from GIT
+=====================
+
+Ubuntu Installation from GIT
+----------------------------
+
+This document will explain how to install and use the most recent code of
+Suricata on Ubuntu. Installing from GIT on other operating systems is
+basically the same, except that some commands are Ubuntu-specific
+(like sudo and apt-get). In case you are using another operating system,
+you should replace those commands with your OS-specific commands.
+
+Pre-installation requirements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Before you can build Suricata for your system, run the following command
+to ensure that you have everything you need for the installation.
+
+.. code-block:: bash
+
+  sudo apt-get -y install libpcre3 libpcre3-dbg libpcre3-dev \
+  build-essential autoconf automake libtool libpcap-dev libnet1-dev \
+  libyaml-0-2 libyaml-dev pkg-config zlib1g zlib1g-dev libcap-ng-dev \
+  libcap-ng0 make libmagic-dev libjansson-dev rustc cargo jq git-core
+
+Add ``${HOME}/.cargo/bin`` to your path:
+
+.. code-block:: bash
+
+  export PATH=$PATH:${HOME}/.cargo/bin
+  cargo install --force cbindgen
+
+Depending on the current status of your system, it may take a while to
+complete this process.
+
+**IPS**
+
+By default, Suricata works as an IDS. If you want to use it as an IDS and IPS
+program, enter:
+
+.. code-block:: bash
+
+  sudo apt-get -y install libnetfilter-queue-dev libnetfilter-queue1 \
+  libnfnetlink-dev libnfnetlink0
+
+Suricata
+~~~~~~~~
+
+First, it is convenient to create a directory for Suricata.
+Name it 'suricata' or 'oisf', for example. Open the terminal and enter:
+
+.. code-block:: bash
+
+  mkdir suricata  # mkdir oisf
+
+Followed by:
+
+.. code-block:: bash
+
+  cd suricata  # cd oisf
+
+Next, enter the following line in the terminal:
+
+.. code-block:: bash
+
+  git clone git@github.com:OISF/suricata.git
```

Comment:
Note that this requires setup of the SSH-key with github; to clone without SSH keys: `git clone https://github.com/OISF/suricata.git`